### PR TITLE
FEATURE: Add maintenance readonly mode that does not share the same k…

### PR DESCRIPTION
**Pull request by tgxworld**
…ey as user triggered readonly.

### CODE REVIEW

File: lib/discourse.rb [hunk 1]
```diff
@@ -215,31 +215,52 @@ def self.base_url
     base_url_no_prefix + base_uri
   end
 
-  def self.enable_readonly_mode
-    $redis.set(readonly_mode_key, 1)
+  MAINTENANCE_READONLY_MODE_KEY ||= "readonly_mode:maintenance".freeze
+  READONLY_MODE_KEY_TTL ||= 60
```
Comment (tgxworld) [inline lib/discourse.rb]: I feel like this TTL is too short to survive a rebuild but I guess we've not gotten bug reports about this yet.

**Comment by tgxworld** [comment]
ping @SamSaffron

File: lib/discourse.rb [hunk 2]
```diff
@@ -215,31 +215,52 @@ def self.base_url
...
+  def self.enable_maintenance_readonly_mode
+    enable_readonly_mode(MAINTENANCE_READONLY_MODE_KEY)
+  end
+
+  def self.disable_maintenance_readonly_mode
+    disable_readonly_mode(MAINTENANCE_READONLY_MODE_KEY)
+  end
+
+  def self.enable_readonly_mode(key = readonly_mode_key)
+    $redis.setex(key, READONLY_MODE_KEY_TTL, 1)
```
Comment (tgxworld) [inline lib/discourse.rb]: @SamSaffron

File: lib/discourse.rb [hunk 3]
```diff
@@ -215,31 +215,52 @@ def self.base_url
     base_url_no_prefix + base_uri
   end
 
-  def self.enable_readonly_mode
-    $redis.set(readonly_mode_key, 1)
```
Comment (tgxworld) [inline lib/discourse.rb]: Readonly key no TTL

**Comment by SamSaffron** [comment]
I am closing this per our discussion

**Pull request closed without merging**